### PR TITLE
Issue #1600/1891228: Stop image_style_deliver from delivering invalid…

### PR DESCRIPTION
**Pull request by serundeputy**
… headers.

### CODE REVIEW

File: core/modules/image/tests/image.test [hunk 1]
```diff
@@ -170,6 +170,22 @@ class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
   }
 
   /**
+   * Test that we do not pass an array to backdrop_add_http_header.
+   */
+  function testImageContentTypeHeaders() {
+    $files = $this->backdropGetTestFiles('image');
+    $file = array_shift($files);
+    // Copy the test file to private folder.
+    $private_file = file_copy($file, 'private://', FILE_EXISTS_RENAME);
```
Comment (quicksketch) [inline core/modules/image/tests/image.test]: file_copy() in Backdrop requires a File entity. As this is an unmanaged file, this needs to be:

```
$private_uri = file_unmanaged_copy($file->uri, 'private://test.png');
```

Comment (serundeputy) [inline core/modules/image/tests/image.test]: thanks.

I've changed, rebased, and pushed back up.

File: core/modules/image/tests/image.test [hunk 2]
```diff
@@ -170,6 +170,22 @@ class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
...
+   */
+  function testImageContentTypeHeaders() {
+    $files = $this->backdropGetTestFiles('image');
+    $file = array_shift($files);
+    // Copy the test file to private folder.
+    $private_uri = file_unimanaged_copy($file->uri, 'private://test.png');
+    // Tell image_module_test module to return the headers we want to test.
+    state_set('image_module_test_invalid_headers', $private_file->uri);
+    // Invoke image_style_deliver so it will try to set headers.
+    $generated_url = image_style_url($this->style_name, $private_file->uri);
```
Comment (quicksketch) [inline core/modules/image/tests/image.test]: Thanks for updating the PR! Though now `$private_file` is missing, so `$private_file->uri` needs to be `$private_uri`.

Comment (serundeputy) [inline core/modules/image/tests/image.test]: updated, rebased pushed :smile:

**Comment by quicksketch** [comment]
Test failure is invalid here. Thanks @serundeputy!

**Pull request merged**